components/config/prefs.rs

```diff
@@ -58,42 +54,13 @@ pub fn pref_map() -> &'static Preferences<'static, Prefs> {
     &PREFS
 }
 
-pub(crate) fn add_user_prefs() {
-    if let Some(path) = user_prefs_path() {
-        init_user_prefs(path);
+pub fn add_user_prefs(prefs: HashMap<String, PrefValue>) {
+    if let Err(error) = PREFS.set_all(prefs.into_iter()) {
+        writeln!(&mut stderr(), "Error setting preference: {:?}", error)
+            .expect("failed printing to stderr");
```

Comment:
Instead of reading userconfig/prefs.json, we  let the embedder decide how to read the user pref. The embedder calls add_user_prefs.